bot/bot.py

```diff
@@ -48,6 +51,19 @@ async def setup_hook(self) -> None:
 
     async def on_error(self, event: str, *args, **kwargs) -> None:
         """Log errors raised in event listeners rather than printing them to stderr."""
+        e_val = exception()
+
+        if isinstance(e_val, Forbidden):
+            message = args[0] if event == "on_message" else args[1] if event == "on_message_edit" else None
+            try:
+                await handle_forbidden_from_block(e_val, message)
+            except Forbidden:
+                # Error wasn't handled, so handle below
+                pass
+            else:
+                # Error was handled, so return
+                return
```

Comment:
I haven't used contextlib in a bit. Would it be acceptable to do something such as

```
    with contextlib.suppress(Forbidden):
        await handle_forbidden_from_block(e_val, message)
        return
```

Reply:
Yeah, I think that would be good.

The behaviour of `handle_forbidden_from_block` is pretty confusing, so I think it would be good to clear things up with a comment.
```python
    with contextlib.suppress(Forbidden):
        # Attempt to handle the error. This reraises the error if's not due to a block, 
        # in which case the error is suppressed and handled normally. Otherwise, it was
        # handled so return.
        await handle_forbidden_from_block(e_val, message)
        return
```

Reply:
[56754a8](https://github.com/python-discord/bot/pull/2580/commits/56754a89c5e9ce3f0528c9abb01ef2b1b7e8be75)